### ISSUE TITLE
Deserializing through an `Option`

### ISSUE THREAD

**Issue by philpax** [ModProg/toml-datetime-compat#1]
Hi there!

Thanks for this library, it's saved me the effort of handling the deserialization myself. I've run into a bit of an edge case: I'm parsing the TOML front matter in a Markdown document, where the date may or may not be present:

```rust
#[derive(Debug, Deserialize)]
pub struct DocumentMetadata {
    pub title: String,
    #[serde(with = "toml_datetime_compat")]
    pub date: Option<chrono::NaiveDate>,
    pub taxonomies: Option<DocumentTaxonomies>,
}
```

However, this doesn't work as the relevant trait isn't implemented for `Option<chrono::NaiveDate>` (or, I suppose, `Option<T>` in general).

I've worked around this by using a tuple struct:
```rust
#[derive(Debug, Deserialize, Clone, Copy)]
#[serde(transparent)]
pub struct NaiveDate(#[serde(with = "toml_datetime_compat")] chrono::NaiveDate);

#[derive(Debug, Deserialize)]
pub struct DocumentMetadata {
    pub title: String,
    date: Option<NaiveDate>,
    pub taxonomies: Option<DocumentTaxonomies>,
}
impl DocumentMetadata {
    pub fn date(&self) -> Option<chrono::NaiveDate> {
        self.date.map(|d| d.0)
    }
}
```

but it might be a good idea to implement something like `impl<T: FromToTomlDateTime> FromToTomlDateTime for Option<T> {}` that will use the standard `serde` `Option` parsing, and then forward to the relevant `FromToTomlDateTime` implementation.

**Comment by ModProg** [comment]
Sounds reasonable, I'll look into this sometime. 

If you want, I'd also accept a contribution making option parsing work, if this doesn't work implicitly a `toml_datetime_compat::option` module could also be the solution.

**Comment by ModProg** [comment]
Thanks for taking it up! Published as `0.3`
